conans/util/files.py

```diff
@@ -48,6 +48,7 @@ def make_file_read_only(file_path):
 
 def set_dirty(folder):
     dirty_file = os.path.normpath(folder) + _DIRTY_FOLDER
+    assert not os.path.exists(dirty_file)
```

Comment:
Every assert with the corresponding message

```suggestion
    assert not os.path.exists(dirty_file), "Folder '{}' is already dirty".format(folder)
```